lib/Interpreter/CppInterOp.cpp

```diff
@@ -1027,61 +1056,78 @@ namespace Cpp {
   }
 
   TCppFunction_t
-  BestTemplateFunctionMatch(const std::vector<TCppFunction_t>& candidates,
+  BestOverloadFunctionMatch(const std::vector<TCppFunction_t>& candidates,
```

Comment:
This function feels very similar to `Sema::BuildCallExpr` can we use that instead and unwrap the result from the call expression? I think the cost would be an AST node allocation per call but we currently do that anyway...

Reply:
Yes, it will but we do that anyway. If you prefer not to do it, then write a comment that this is copied from there. FWIW, that did not seem the case when I looked...